evidence/pool.go

```diff
@@ -92,43 +105,81 @@ func (evpool *Pool) Update(block *types.Block, state sm.State) {
 
 	// remove evidence from pending and mark committed
 	evpool.MarkEvidenceAsCommitted(block.Height, block.Time, block.Evidence.Evidence)
+
+	evpool.cleanupValToLastHeight(block.Height)
```

Comment:
`ValToLastHeight` also needs to be updated in the sense that the validators of the the new block should be added to this

Reply:
right! good catch